From b1dec33ddb4f0a9fe16a95fe07e4b30efec51fb9 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 12 Jan 2022 14:11:06 +0200
Subject: [PATCH] [SW-0] gaudi: preboot: add image size validation

Preboot binary image size must not exceed 128kb (2 sectors of flash).
Image size validation added in makefile

Change-Id: Ic1a34d4c6ee016ff6d30bee51fb1a85b09eab0e3
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi/arm/Makefile b/arch/gaudi/arm/Makefile
index f01c94e..ea14dba 100644
--- a/arch/gaudi/arm/Makefile
+++ b/arch/gaudi/arm/Makefile
@@ -1,5 +1,6 @@
 FW_SIZE := 0x4000
 PPBOOT_MAX_SIZE := 20480
+PREBOOT_MAX_SIZE := 131072
 
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/pll.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/security.c)
@@ -13,3 +14,4 @@
 CFLAGS += -mstrict-align
 
 image_validation: image_validation_common
+image_validation: image_validation_preboot
